packages/didcomm/src/transport/queue/QueueTransportRepository.ts

```diff
@@ -0,0 +1,18 @@
+import { AgentContext } from '@credo-ts/core'
+import type {
+  AddMessageOptions,
+  GetAvailableMessageCountOptions,
+  RemoveMessagesOptions,
+  TakeFromQueueOptions,
+} from './QueueTransportRepositoryOptions'
+import type { QueuedMessage } from './QueuedMessage'
+
+export interface QueueTransportRepository {
```

Comment:
Should we take into account an agent that does not want to queue? Can we disable it instead of using in memory (which could be a memory leak)



also should we add some extra metadata to the queued messgages? So like a state (awaiting confirmation, or the date/time since we are awaiting confirmation) or do we see this more as a repository specific thing to ha e?